packages/grafana-ui/src/components/Forms/FieldSet.mdx

```diff
@@ -0,0 +1,42 @@
+import { Story, Preview, Props } from '@storybook/addon-docs/blocks';
+import { FieldSet } from './FieldSet';
+
+# FieldSet
+
+Component used to group elements inside a form, equivalent to HTML's [fieldset](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset) tag.
+Accepts optional label, which, if provided, is used as text for the set's legend.
+
+
+### Usage
+
+```jsx
+import { FieldSet } from '@grafana/ui';
+
+<Form onSubmit={() => console.log('Submit')}>
+  {() => (
+    <>
+      <FieldSet label="Details">
```

Comment:
When I look at this now I think `label` prop makes a lot of sense, takin `Field` API into consideration 👍